ethcore/src/snapshot/service.rs

```diff
@@ -260,6 +297,37 @@ impl Service {
 		Ok(())
 	}
 
+	/// Take a snapshot at the block with the given number.
+	/// calling this while a restoration is in progress or vice versa
+	/// will lead to a race condition where the first one to finish will
+	/// have their produced snapshot overwritten.
+	pub fn take_snapshot(&self, client: &Client, num: u64) -> Result<(), Error> {
+		info!("Taking snapshot at #{}", num);
```

Comment:
Yep. If that's non-trivial I'd be happy with just the completion message. Would be useful for testing